docs/topics/testing/tools.txt

```diff
@@ -1683,15 +1683,17 @@ your test suite.
 
     Output in case of error can be customized with the ``msg`` argument.
 
-.. method:: TransactionTestCase.assertQuerysetEqual(qs, values, transform=repr, ordered=True, msg=None)
+.. method:: TransactionTestCase.assertQuerysetEqual(qs, values, transform=None, ordered=True, msg=None)
 
-    Asserts that a queryset ``qs`` returns a particular list of values ``values``.
+    Asserts that a queryset ``qs`` matches a particular iterable of values
+    ``values``.
 
-    The comparison of the contents of ``qs`` and ``values`` is performed by
-    applying ``transform`` to ``qs``. By default, this means that the
-    ``repr()`` of each value in ``qs`` is compared to the ``values``. Any other
-    callable can be used if ``repr()`` doesn't provide a unique or helpful
-    comparison.
+    If ``transform`` is supplied as a function, ``values`` is compared to a
+    list produced by applying ``transform`` to each member of ``qs``.
+
+    If ``transform`` is not supplied and ``values`` is a list of string
+    elements, ``values`` is compared to a list produced by applying ``repr()``
+    to each member of ``qs``.
```

Comment:
Also, since this is going to get a lot of eyes when folks rewrite their tests, how about we clarify the third possibility, which is the new convenience: direct comparison.

I'm not 100% sure of the wording, but something like: "Otherwise, without a ``transform``, the equality of each member of ``qs`` and ``values`` is asserted directly."